docs/bokeh/source/docs/user_guide/interaction/js_callbacks.rst

```diff
@@ -82,28 +82,120 @@ JavaScript code to run in the browser when an event occurs.
 
     from bokeh.models.callbacks import CustomJS
 
-    callback = CustomJS(args=dict(xr=plot.x_range), code="""
+    callback = CustomJS(args=dict(xr=plot.x_range, yr=plot.y_range, slider=slider), code="""
+    // imports
+    import {some_function, SOME_VALUE} from "https://cdn.jsdelivr.net/npm/package@version/file"
 
-    // JavaScript code goes here
+    // constants, definitions and state
+    const MY_VALUE = 3.14
 
-    const a = 10;
+    function my_function(value) {
+        return MY_VALUE*value
+    }
 
-    // the model that triggered the callback is cb_obj:
-    const b = cb_obj.value;
+    class MyClass {
+        constructor(value) {
+            this.value = value
+        }
+    }
+
+    let count = 0
+
+    // the callback function
+    export default (args, obj, data, context) => {
+        count += 1
+        console.log(`CustomJS was called ${count} times`)
 
-    // models passed as args are automagically available
-    xr.start = a;
-    xr.end = b;
+        const a = args.slider.value
+        const b = obj.value
 
+        const {xr, yr} = args
+        xr.start = my_function(a)
+        xr.end = b
+    }
     """)
 
-Note that in addition to the ``code`` property, ``CustomJS`` also accepts
-an ``args`` property that maps string names to Bokeh models. Any Bokeh
-models that are configured in ``args`` (on the "Python side") will
-automatically be available to the JavaScript code by the corresponding name.
+The code snippet must contain a default export, which must be a function defined
+either using arrow function syntax ``() => {}`` or a classical function syntax
+ ``function() {}``. Depending on the context, this function may be an async
+function, a generator function or an async generator function. Also depending
+on the context, this function may or may not need to return a value.
+
+The callback function uses four positional arguments:
+
+* ``args`` - this maps to ``CustomJS.args`` property, allowing for mapping
+    names to serializable values, typically providing access to Bokeh models
+    from the code snippet.
+* ``obj`` - this refers to the model that emitted the callback (this is
+    the model that the callback is attached to).
+* ``data`` - this is a mapping between names and values provided by the
+    emitter of this callback. This depends on the caller, the event and
+    possibly the context in which the event occurred. For example, a select
+    tool will use ``data`` to provide selection geometry, among other things.
+* ``context`` - this is an additional broader context provided by bokehjs,
+    which is a mapping between names and values, similarly to ``data``.
+    Currently only ``index`` is provided, which allows the user to access
+    bokehjs' view index.
```

Comment:
```suggestion
* ``args``
    this maps to ``CustomJS.args`` property, allowing for mapping
    names to serializable values, typically providing access to Bokeh models
    from the code snippet.
* ``obj``
    this refers to the model that emitted the callback (this is
    the model that the callback is attached to).
* ``data``
    this is a mapping between names and values provided by the
    emitter of this callback. This depends on the caller, the event and
    possibly the context in which the event occurred. For example, a select
    tool will use ``data`` to provide selection geometry, among other things.
* ``context``
    this is an additional broader context provided by bokehjs,
    which is a mapping between names and values, similarly to ``data``.
    Currently only ``index`` is provided, which allows the user to access
    bokehjs' view index.
```

|**current**|**suggested**|
|--|--|
|![grafik](https://github.com/bokeh/bokeh/assets/68053396/9b921091-bb7c-42e9-99fb-be9f4e915a96)|![grafik](https://github.com/bokeh/bokeh/assets/68053396/f78378ac-ae57-416d-9f21-65bb627cd10b)|

This avoids that some of the explanation is bold